src/main/java/yuquiz/domain/notification/repository/NotificationRepository.java

```diff
@@ -11,4 +14,12 @@ public interface NotificationRepository extends JpaRepository<Notification, Long
     Page<Notification> findAllByUserAndIsChecked(User user, boolean isChecked, Pageable pageable);
 
     Page<Notification> findAllByUser(User user, Pageable pageable);
+
+    @Modifying
+    @Query(value = "UPDATE Notification " +
+            "SET isChecked = true " +
+            "WHERE id IN :notifications AND user_id = :userId",
+            nativeQuery = true)
+    void readNotificationsWithValidation(@Param("notifications") Long[] notifications,
+                                              @Param("userId") Long userId);
```

Comment:
문제되는 사항은 아니지만 아래의 코드처럼 JPQL을 활용할 수 있을 것 같은데 nativeQuery를 사용하신 이유가 있으신가용?

```suggestion
    @Query(value = "UPDATE Notification n " +
           "SET n.isChecked = true " +
           "WHERE n.id IN :notifications AND n.user.id = :userId")
    void updateNotificationsToChecked(@Param("notifications") List<Long> notifications, 
                                      @Param("userId") Long userId);
```

Reply:
뭐 딱히 큰 이유는 없고
그냥 nativeQuery를 한번 써보고 싶었다랄까요..?ㅎㅎ
근데 지금 생각하니 통일성을 위해 jpql로 변경하는게 좋을 것 같네요